independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/JarClassPathElement.java

```diff
@@ -182,8 +182,8 @@ public boolean isDirectory() {
                 }
                 //multi release jars can add additional entries
                 if (JarFiles.isMultiRelease(jarFile)) {
-                    Set<String> copy = new HashSet<>(paths);
-                    for (String i : copy) {
+                    String[] copyToIterate = paths.toArray(new String[0]);
```

Comment:
I'm wondering if `paths` can contain duplicates and if that is the reason a `Set` was being used

Reply:
Sure it might happen and it still works..  N.B. the `paths` already is a Set; it's being copied only to iterate on it safely as we might make more changed to the set. Nothing beats an array when all you need is iterating - especially as it was re-computing all hashes again.

Reply:
I mean, that HashSet copy is being hot on CPU usage here - for no good reason. Allocations of hashmap entries are also high.

Reply:
Ok cool, I didn't look at the surrounding code, just the changes :)